csharp-advanced-topics/ActionAndFuncDelegate/Tests/Tests.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace Tests
```

Comment:
You seem to have a good hang of writing tests. In the future, if we decide we should go forward, you can use the examples from the tests in your code snippets to avoid repeating the code both in the main app and in the tests. Asserts are a pretty graphic and nice way to show what results are expected. Even better than console writeline.